### ISSUE TITLE
Adding edge with any operator yields `KeyError` when running the simulation

### ISSUE THREAD

**Issue by jajcayn** [pyrates-neuroscience/PyRates#11]
I'd showcase on an example:
consider your Jansen-Rit Circuit implementation from `model_templates`:
```yaml
...
nodes:
    PC: PC
    EIN: EIN
    IIN: IIN
  edges:
    - [PC/PRO/m_out, IIN/RPO_e/m_in, DummyEdge, {weight: 33.75}]
    - [PC/PRO/m_out, EIN/RPO_e/m_in, DummyEdge, {weight: 135.}]
    - [EIN/PRO/m_out, PC/RPO_e_pc/m_in, DummyEdge, {weight: 108.}]
    - [IIN/PRO/m_out, PC/RPO_i/m_in, DummyEdge, {weight: 33.75}]
```

Now, for the purpose of building more complex edges in the future, let just create linear coupling operator, assign it to the edge and scale the edge not via weight but rather via coupling value (the results should be the same):
```yaml
LinearCoupling:
  base: OperatorTemplate
  equations: "V_in = c * V_out"
  variables:
    V_in:
      default: output
    c:
      default: 1.
    V_out:
      default: input

LinearEdge:
  base: EdgeTemplate
  operators:
    - LinearCoupling
```
and build the JRC with this edge template as:
```yaml
 nodes:
    PC: PC
    EIN: EIN
    IIN: IIN
  edges:
    - [PC/PRO/m_out, IIN/RPO_e/m_in, LinearEdge, { LinearCoupling/c: 33.75, weight: 1. }]
    - [PC/PRO/m_out, EIN/RPO_e/m_in, LinearEdge, { LinearCoupling/c: 135., weight: 1. }]
    - [EIN/PRO/m_out, PC/RPO_e_pc/m_in, LinearEdge, { LinearCoupling/c: 108., weight: 1. }]
    - [IIN/PRO/m_out, PC/RPO_i/m_in, LinearEdge, { LinearCoupling/c: 33.75, weight: 1. }]
```
.

As you can see, the coupling parameter `c` takes the value of the edge weight and with weight being 1., this JRC circuit should have the same output as the original one from model template.

However, running this circuit (in the tutorial notebook) yields an error:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/circuit.py in get_node_var(self, key, apply_idx)
    843         try:
--> 844             return self[key]
    845         except KeyError:

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/abc.py in __getitem__(self, key)
     77             else:
---> 78                 raise e
     79 

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/abc.py in __getitem__(self, key)
     70             key = next(key_iter)
---> 71             item = self.getitem_from_iterator(key, key_iter)
     72             for key in key_iter:

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/circuit.py in getitem_from_iterator(self, key, key_iter)
    379         else:
--> 380             item = self.graph.nodes[key]["node"]
    381 

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/networkx/classes/reportviews.py in __getitem__(self, n)
    177     def __getitem__(self, n):
--> 178         return self._nodes[n]
    179 

KeyError: 'PC'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-12-ddc332121c49> in <module>
      5                                      'V_EIN': 'EIN/RPO_e/PSP',
      6                                      'V_IIN': 'IIN/RPO_e/PSP'}, 
----> 7                             inputs={'PC/RPO_e_pc/u': ext_input})

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/circuit.py in run(self, simulation_time, inputs, outputs, sampling_step_size, out_dir, verbose, profile, **kwargs)
    966 
    967                 # extract respective output variables from the network and store their information
--> 968                 for var_key, var_info in self.get_node_var(val).items():
    969                     var_shape = tuple(var_info['var'].shape)
    970                     if var_shape in output_shapes:

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/circuit.py in get_node_var(self, key, apply_idx)
    849 
    850             # split original node keys
--> 851             node_keys = [key.split('/') for key in self.label_map]
    852 
    853             # remove all nodes from original node keys that are not referred to

~/.virtualenvs/sleepy3/lib/python3.7/site-packages/pyrates-0.8.0-py3.7.egg/pyrates/ir/circuit.py in <listcomp>(.0)
    849 
    850             # split original node keys
--> 851             node_keys = [key.split('/') for key in self.label_map]
    852 
    853             # remove all nodes from original node keys that are not referred to

AttributeError: 'tuple' object has no attribute 'split'
```

The `label_map` property of `CircuitIR` contains nodes' labels (`PC`, `EIN`, `IIN`) and edges in the form `(PC, EIN, 0)` and that is exactly the culprit - it fails on `split()` when encountering edge.

**Comment by Richert** [comment]
This issue is related to #7. EdgeOperators are currently not correctly handled, after a change that was made to way how edges are handled in the equation system in the backend. We are working on re-enabling them though. In the mean-time, I recommend to implement operators at the node level when possible. Usually, in systems that use a Jansen-Rit like formalism, there exist multiple valid representations in PyRates. For example, have a look at 
`model_templates.jansen_rit.simple_jansenrit.yaml`
in there, we have different YAML templates that can all represent the Jansen-Rit circuit. In one of them, a single Node with a single operator is used and all edges are directly incorporated into the equations. I suggest to have a look at these examples and move the differential equations of the edge to the node, such that the edge can be reduced to a scalar weight and delay.

I hope this helps and I will keep you updated on the progress regarding the re-enabling of the _CouplingOperator_. 

Cheers, Richard

**Comment by dafrose** [comment]
Hi Nikola, this issue should be solved with the next version (0.9.0). If you want, you can check out the latest version here https://github.com/pyrates-neuroscience/PyRates/commit/f246f0eb95f62fc5cd5ef439dab94acdf49f506f and try for yourself. 

Best,
Daniel

**Comment by Richert** [comment]
closing this now. This has been resolved in PyRates 0.9.0
